Platform: correct `posix_spawn` API nullability

compnerd opened the pull request:
The `posix_spawn` API is not annotated appropriately for nullability. Adjust the signature to allow proper usage with older NDKs.

compnerd committed:
Platform: correct `posix_spawn` API nullability

The `posix_spawn` API is not annotated appropriately for nullability.
Adjust the signature to allow proper usage with older NDKs.

compnerd commented:
@swift-ci please smoke test

grynspan commented:
Looks good (as discussed). Please update `posix_spawn_file_actions_init()` too as we discussed offline. Thanks!

@finagolfin @marcprux FYI.

compnerd committed:
Platform: correct `posix_file_actions_init` on Android

Adjust the API annotations to get this imported into Swift properly. The
outer pointer may be `nullptr`, but the inner pointer may not.

compnerd commented:
@swift-ci please smoke test

compnerd commented:
@swift-ci please smoke test Windows platform

finagolfin commented:
I don't think we should support older NDKs that didn't have these annotations, but as long as this won't break anything in the current LTS NDK 27, don't see a problem with it.
